models/forum.server.model.ts

```diff
@@ -0,0 +1,163 @@
+      comments = ['']
+
+    // Create new forum post document
+    const newPost = new Forum({
+        userID,
+        communityID,
+        title,
+        bodyText,
+        edited,
+        upVotes,
+        downVotes,
+        attachments,
+        comments
+    })
+
+    // Save new forum post document to database collection
+    newPost.save()
+      .then((res) => {
+          return done(res)
+      })
+      .catch((err) => {
+          // Forum post is already in the database with unique attributes, return duplicate conflict error
+          if (err.code === 11000) {
+              return done({ err: 'Conflict', status: 409 })
+          }
+          // Any other database error, return internal server error
+          return done({ err: 'Internal server error', status: 500 })
+      })
+}
+
+/**
+ * Search for a forum post in the database
+ * @param id forum post ID
+ * @param done function callback, returns status code, and message if error, or JSON if successful
+ */
+export function searchPostById (id, done) {
+    try {
+        Forum.findById(id)
+          .then((res) => done(res))
+          .catch((err) => {
+              return done({ status: 404, err: err })
+          })
+    } catch (err) {
+        return done({ err: 'Internal server error', status: 500 })
+    }
+}
+
+/**
+ * Delete an existing forum post matching a given ID
+ * @param id the ID for matching to the database document being deleted
+ * @param done function callback, returns status code and message if error
+ */
+export function deletePostById (id, done) {
+    Forum.deleteOne({ _id: id })
+      .then((res) => {
+          if (res.deletedCount === 0) {
+              return done({ err: 'Not found', status: 404 })
+          }
+          return done(res)
+      })
+      .catch((err) => {
+          return done({ err: 'Internal server error', status: 500 })
+      })
+}
+
+/**
+ * Updates given fields of a database collection document matching a given ID
+ * @param id the ID of the document being updated
+ * @param updates the document field(s) being updated
+ * @param done function callback, returns status code and message if error
+ */
+export function updatePostById (id, updates, done) {
+    // Search for a document matching the given ID to increment upVotes and downVotes
+    searchPostById(id, function (result) {
+        if (result.err) {
+            // Return the error message with the error status
+            return done(result)
+        } else {
+            updates['upVotes'] = updates.upVotes ? updates.upVotes + result.upVotes : result.upVotes
+            updates['downVotes'] = updates.downVotes ? updates.downVotes + result.downVotes : result.downVotes
+            updates['comments'] = updates.comments ? result.comments.concat(updates.comments) : result.comments
+
+            // If the update does not just involve up votes, down votes, or comments, and actual edits
+            if (Object.keys(updates).length > 3) {
```

Comment:
Can we make this magic number a constant

Reply:
Let's not focus on all the unrelated issues right now, there are more than enough of them to fix later. If we try fix them now we cant parallelise efficently